### PR TITLE
PR: Add 'Show in external file explorer' option

### LINKED ISSUE #2628: Show current file in file browser
It would be nice to have a menu item to show the current file in the file explorer (move to directory, select file), ideally in the context menu.

### PR DESCRIPTION
**Pull request by impact27**
Fixes #2628

### CODE REVIEW

**Comment by goanpeca** [comment]
@impact27 thanks for your contribution.

could you add the `Fixes #NNN` issue on the description? If there is no available issue open to relate this to, could you create the issue?

**Comment by ccordoba12** [comment]
@impact27, please merge or rebase with the latest changes in our `3.x` branch to fix the errors in CircleCI.

**Comment by impact27** [comment]
There is #2628 but this is only for OSX

**Comment by impact27** [comment]
I added the same option to the tabs in the editor. I had to redefine findInFinder as I needed access to self.get_current_filename()

**Comment by impact27** [comment]
For linux "xdg-open folder" works but I don't know how to select a specific file. No ideas how to do that in windows.

**Comment by impact27** [comment]
Could somebody test that with windows and linux? It works on mac

File: spyder/widgets/editor.py [hunk 1]
```diff
@@ -430,6 +443,16 @@ def __init__(self, parent, actions):
         #For opening last closed tabs
         self.last_closed_files = []
 
+    @Slot()
+    def showInFinder(self, fnames=None):
```
Comment (ccordoba12) [inline spyder/widgets/editor.py]: Please rename this method to `show_in_external_file_explorer`

File: spyder/widgets/editor.py [hunk 2]
```diff
@@ -359,9 +360,21 @@ def __init__(self, parent, actions):
...
-                                       copy_to_cb_action, None, close_right,
+                                       copy_to_cb_action]
+        
+        if sys.platform == 'darwin':
+            show_in_finder_action = create_action(self, _("Show in finder"), 
+                                              triggered=self.showInFinder)
+        else:
+            show_in_finder_action = create_action(self, 
+                                          _("Show in external file explorer"), 
+                                              triggered=self.showInFinder)
```
Comment (ccordoba12) [inline spyder/widgets/editor.py]: You don't need to create two actions. Only one action is needed and you change its text according to the operating system, i.e. something like:

```python
if sys.platform == 'darwin':
    text = _("Show in finder")
else:
    text = _("Show in external file explorer")
external_fileexp_action = create_action(self, text,
                            triggered=self.show_in_external_file_explorer)
```

File: spyder/widgets/editor.py [hunk 3]
```diff
@@ -359,9 +360,21 @@ def __init__(self, parent, actions):
         close_all_but_this = create_action(self, _("Close all but this"),
                                            triggered=self.close_all_but_this)
 
-        self.menu_actions = actions + [None, fileswitcher_action,
+        actions = actions + [None, fileswitcher_action,
                                        symbolfinder_action,
-                                       copy_to_cb_action, None, close_right,
+                                       copy_to_cb_action]
```
Comment (ccordoba12) [inline spyder/widgets/editor.py]: Please leave this as it was, and instead move the creation of `external_fileexp_action` before this.

File: spyder/widgets/explorer.py [hunk 4]
```diff
@@ -42,7 +44,16 @@
 except:
     nbexporter = None    # analysis:ignore
 
-
+def open_file_external_explorer(filename):
```
Comment (ccordoba12) [inline spyder/widgets/explorer.py]: Please change this to `open_file__in_external_explorer`

File: spyder/widgets/explorer.py [hunk 5]
```diff
@@ -293,6 +304,15 @@ def create_file_manage_actions(self, fnames):
             actions.append(edit_action)
         else:
             actions.append(open_action)
+        if sys.platform == 'darwin':
+            show_in_finder_action = create_action(self, _("Show in finder"), 
+                                              triggered=self.showInFinder)
+        else:
+            show_in_finder_action = create_action(self,
+                                        _("Show in external file explorer"), 
+                                              triggered=self.showInFinder)
```
Comment (ccordoba12) [inline spyder/widgets/explorer.py]: As I said above, you don't need to create two actions here, just one.

File: spyder/widgets/explorer.py [hunk 6]
```diff
@@ -606,6 +626,16 @@ def rename_file(self, fname):
                               ) % (osp.basename(fname), to_text_string(error)))
 
     @Slot()
+    def showInFinder(self, fnames=None):
```
Comment (ccordoba12) [inline spyder/widgets/explorer.py]: Again, rename this to `show_in_external_file_explorer`

**Comment by ccordoba12** [comment]
@impact27, I did an initial review to your work. Once we finish reviewing your code, we'll test it on Linux and Windows.

**Comment by ccordoba12** [comment]
Thanks for the changes @impact27! Could you merge again with our `3.x` branch to fix another error in CircleCI? Thanks!

**Comment by rlaverde** [comment]
I refactor to delete some duplicated code and make it more reusable, and merge it with upstream/3.x

File: spyder/widgets/explorer.py [hunk 7]
```diff
@@ -42,6 +44,26 @@
 except:
     nbexporter = None    # analysis:ignore
 
+def open_file_in_external_explorer(filename):
+    if sys.platform == "darwin":
+        subprocess.call(["open", "-R", filename])
+    else:
+        filename=os.path.dirname(filename)
+        if sys.platform == "win32":
```
Comment (ccordoba12) [inline spyder/widgets/explorer.py]: Please change this to

    if os.name == 'nt':

That's how we check for Windows in all our codebase

File: spyder/widgets/explorer.py [hunk 8]
```diff
@@ -293,6 +315,13 @@ def create_file_manage_actions(self, fnames):
             actions.append(edit_action)
         else:
             actions.append(open_action)
+        if sys.platform == 'darwin':
+            text=_("Show in finder")
```
Comment (ccordoba12) [inline spyder/widgets/explorer.py]: `finder` -> `Finder`

**Pull request merged**